standard/array.lua

```diff
@@ -350,6 +350,20 @@ function Array.extractValues(tbl)
 	return values
 end
 
+---@comment Extracts values from a given table into an array in a sorted way
+---@generic K, V, I
+---@param tbl {[K]: V}
+---@param order? fun(tbl: {[K]: V}, key1: K, key2: K): boolean
+---@return {[I]: V}
+
+function Array.extractValuesInOrder(tbl, order)
+	local array = {}
+	for _, item in Table.iter.spairs(tbl, order) do
```

Comment:
```suggestion
function Array.extractValues(tbl, iterator, ...)
	iterator = iterator or pairs
	local array = {}
	for _, item in iterator(tbl, ...) do
```

Could make it more generic and replace the current extractValues

Reply:
Would we also want to do the same for `extractKeys`